### PR TITLE
[REG2.068a] Issue 14844 - __traits(allMembers) must not visit yet not instantiated template members

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=14844

### CODE REVIEW

File: src/traits.c [hunk 1]
```diff
@@ -945,22 +945,21 @@ Expression *semanticTraits(TraitsExp *e, Scope *sc)
...
-            sds = import->mod;
+            s = imp->mod;
         }
-        else if ((sds = s->isScopeDsymbol()) == NULL)
+        ScopeDsymbol *sds = s->isScopeDsymbol();
+        if (!sds || sds->isTemplateDeclaration())
         {
             e->error("%s %s has no members", s->kind(), s->toChars());
-            goto Lfalse;
+            return new ErrorExp();
```
Comment (MartinNowak) [inline src/traits.c]: Why did you introduce the `ErrorExp`, seems unrelated and is inconsistent with the `goto Lfalse` above.

**Comment by MartinNowak** [comment]
Rest LGTM.

File: src/traits.c (hunk 1 above)
Comment (9rnsr) [inline src/traits.c]: It was just for the redundant error message. I think if any errors happen in `semanticTraits`, it should return `ErrorExp`. But it's not important point of this PR, so I tweaked test case, and then changed to use `goto Lfalse` in here.

**Comment by 9rnsr** [comment]
Thanks.
